cmd/api.go

```diff
@@ -0,0 +1,82 @@
+package cmd
+
+import (
+	"bufio"
+	"crypto/tls"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"time"
+)
+
+// Reads a file given its filename and returns a list containing each of its lines.
+func parseFile(filename string) (lines []string, err error) {
+
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	scanner := bufio.NewScanner(file)
+	scanner.Split(bufio.ScanLines)
+
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+
+	err = file.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	return lines, nil
+}
+
+type header struct {
+	key   string
+	value string
+}
+
+// Makes a GET request using headers `headers` proxy `proxy`.
+//
+// Uses a custom method if specified.
+func request(method, uri string, headers []header, proxy *url.URL) (statusCode int, response []byte, err error) {
+
+	if method == "" { // check if it is nil
+		method = "GET"
+	}
+
+	var _proxy *url.URL
+	if len(proxy.Host) != 0 {
+		_proxy = proxy
+	} else {
+		_proxy = nil
+	}
```

Comment:
Made this to avoid:

```go
client := &http.Client{Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
          DialContext: (&net.Dialer{Timeout: 6 * time.Second}).DialContext}}

if len(proxy.Host) != 0 {
  client = &http.Client{Transport: &http.Transport{Proxy: http.ProxyURL(proxy),
	  TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, DialContext: (&net.Dialer{Timeout: 6 * time.Second}).DialContext}}
}
```

Since `http.ProxyURL` returns `nil` for a `nil` argument, and as per `http.Transport`'s `Proxy` documentation: `If Proxy is nil or returns a nil *URL, no proxy is used`